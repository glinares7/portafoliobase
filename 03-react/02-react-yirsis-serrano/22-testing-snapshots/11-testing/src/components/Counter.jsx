import React from "react";
import { useCounter } from "../hooks/useCounter";
import ButtonCounter from "./ButtonCounter";

const Counter = () => {
  //* Refactorizando las funciones con testing en una sola agregando variables

  const { counter, handleModifyCounter, handleReset } = useCounter();

  // const handleDis = () => {
  //   setCounter(counter - 1);
  // };

  // <h2 data-testid="counter">Counter: {counter}</h2>
  return (
    <>
      <h2 role="counter">Counter: {counter}</h2>

      <ButtonCounter name="aumentar" action={handleModifyCounter} />

      <button aria-label="reset" onClick={handleReset}>
        Reset
      </button>

      <ButtonCounter name="disminuir" action={handleModifyCounter} value={-1} />
    </>
  );
};

export default Counter;
