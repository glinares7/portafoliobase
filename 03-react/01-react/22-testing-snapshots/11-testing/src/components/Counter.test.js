import { fireEvent, render, screen } from "@testing-library/react";
import Counter from "./Counter";
import userEvent from "@testing-library/user-event";

//* agrupador de pruebas
describe("<Counter />", () => {
  beforeEach(() => {
    render(<Counter />);
  });

  // beforeAll(() => {
  //   render(<Counter />);
  // });
  // afterEach(() => {
  //   userEvent.click(screen.getByLabelText("reset"));
  // });

  it("Primer snapshot", () => {
    expect(screen.getByRole("counter")).toMatchSnapshot();
  });
  it("Prueba de que el titulo y el estado se renderiza corectamente", () => {
    //*   macher esperas que ocurra
    //* valor actual
    // expect(wrapper.getByText("Counter: 0").tagName).toBe("H2");
    expect(screen.getByText("Counter: 0").tagName).toBe("H2");
  });

  it("Verificar que el click + 1 aumente corectamente", () => {
    fireEvent.click(screen.getByLabelText("aumentar"));

    expect(screen.getByRole("counter").textContent).toContain("Counter: 1");
    // console.log(screen.getByText("+1").tagName);
  });

  it("Verificar que el click -1 disminuya corectamente", () => {
    // textField."adivinando"

    // userEvent

    userEvent.click(screen.getByLabelText("disminuir"));

    expect(screen.getByRole("counter").textContent).toContain("Counter: -1");
    // expect(screen.getByTestId("counter").textContent).toContain("Counter: 1");
    // console.log(screen.getByText("+1").tagName);
  });
  //   it("Prueba que 1 + 1 es 2 y no es 3", () => {
  //     expect(1 + 1).toBe(2);
  //     expect(1 + 1).not.toBe(3);
  //   });

  it("Simular que alguien este usando la app", () => {
    const btnAdd = screen.getByLabelText("aumentar");
    const btnDis = screen.getByLabelText("disminuir");

    userEvent.click(btnAdd);
    userEvent.click(btnAdd);
    userEvent.click(btnAdd);
    userEvent.click(btnAdd);
    userEvent.click(btnAdd);
    userEvent.click(btnAdd);
    userEvent.click(btnAdd);

    expect(screen.getByRole("counter").textContent).toContain("Counter: 7");

    userEvent.click(btnDis);
    userEvent.click(btnDis);
    userEvent.click(btnDis);
    userEvent.click(btnDis);
    userEvent.click(btnDis);
    userEvent.click(btnDis);
    userEvent.click(btnDis); //0
    userEvent.click(btnDis);
    userEvent.click(btnDis);
    userEvent.click(btnDis);
    userEvent.click(btnDis);

    expect(screen.getByRole("counter").textContent).toContain("Counter: -4");
  });

  it("Reset Button", () => {
    const btnAdd = screen.getByLabelText("aumentar");

    userEvent.click(btnAdd);
    userEvent.click(btnAdd);
    userEvent.click(btnAdd);
    userEvent.click(btnAdd);
    userEvent.click(btnAdd);
    userEvent.click(btnAdd);
    userEvent.click(btnAdd);

    userEvent.click(screen.getByLabelText("reset"));
    expect(screen.getByRole("counter").textContent).toContain("Counter: 0");
  });
});
