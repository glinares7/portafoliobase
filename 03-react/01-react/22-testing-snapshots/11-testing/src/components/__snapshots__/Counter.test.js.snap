// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> Primer snapshot 1`] = `
<h2
  role="counter"
>
  Counter: 
  0
</h2>
`;
